Android.gitignore

```diff
@@ -1,80 +1,27 @@
-# Built application files
-*.apk
-*.ap_
-*.aab
-
-# Files for the ART/Dalvik VM
-*.dex
-
-# Java class files
-*.class
-
-# Generated files
-bin/
-gen/
-out/
-
 # Gradle files
 .gradle/
 build/
 
 # Local configuration file (sdk path, etc)
 local.properties
 
-# Proguard folder generated by Eclipse
-proguard/
-
 # Log Files
 *.log
 
-# Android Studio Navigation editor temp files
+# Android Studio generated files and folders
 .navigation/
-
-# Android Studio captures folder
 captures/
+.externalNativeBuild
+*.apk
+output.json
 
 # IntelliJ
 *.iml
-.idea/workspace.xml
-.idea/tasks.xml
-.idea/gradle.xml
-.idea/assetWizardSettings.xml
-.idea/dictionaries
-.idea/libraries
-.idea/caches
-# Android Studio 3 in .gitignore file.
-.idea/caches/build_file_checksums.ser
-.idea/modules.xml
+.idea/
```

Comment:
This is not correct. Rationale: https://github.com/github/gitignore/pull/2527#issuecomment-347049128

Reply:
None of the files inside of `.idea/` are (or should be) required to clean build a fresh clone. Android Studio does not require any of these either. They are auto-generated once opened in Android Studio, which will take information from `build.gradle` (which is in source control).

Reply:
Some configuration files in `.idea` dir are meant to be shared with team members as project-wide IDE configurations. Those configs are not in `build.gradle` so ignoring entire `.idea` is not ideal.

https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems

Reply:
I don't disagree with you. If someone explicitly _wants_ to add certain files to source control, they always have the option to do that. In almost all other cases, none of the files are needed, and are auto-generated by the IDE.   This is true for most Maven/Gradle based projects (which is also mentioned in the links you posted), and especially true for Android Studio.
That's why 9 out of 10 Android projects don't have these files/folders committed to source control. And for those that do, it was often done without explicit intent of the original committer.

Reply:
I don't have any strong feelings on what sort of things under `.idea` should be used, but in situations like this I have some guidance:

 - "What things should be included in version control?" - if a rule is unnecessarily broad, it might interfere with some files that users can commit to the repository and share with others. As `Global/JetBrains.gitignore` shows, there are lots of files in IntelliJ that can be committed, and I've pushed back on PRs in the past where language-specific templates ignore this entire directory.
 - "What things would a typical user expect to be included in version control?" - for more opinionated tooling (which I feel is what Android Studio is doing that @friederbluemle is trying to explain) it may be that these files are unnecessary noise for most users.

If it were me, I'd lean towards being more explicit about what can be committed, but I'll leave it to others to settle on what should be in this template.

Reply:
Yes, that's correct, for other IntelliJ/JetBrains products, some information might be required from the `.idea/` folder. For projects that use the Android Gradle pluin, all information about the build system is contained in `build.gradle`, as explained above. I also added it to the PR description.